### PR TITLE
Adding lock to pickled file

### LINKED ISSUE #1092: Pickle data was truncated
--------
Defect Description
--------
While unpickle the same pk file in RAVEN it might only get the part of the pickled data.
##### What did you see instead?
```
   File "~/projects/raven/framework/Driver.py", line 285, in <module>
     raven()
   File "~/projects/raven/framework/Driver.py", line 238, in raven
     simulation.run()
   File "~/projects/raven/framework/Simulation.py", line 780, in run
     stepInstance.takeAstep(stepInputDict)
   File "~/projects/raven/framework/Steps.py", line 315, in takeAstep
     self._localTakeAstepRun(inDictionary)
   File "~/projects/raven/framework/Steps.py", line 992, in _localTakeAstepRun
     unpickledObj = pickle.load(open(fileobj.getAbsFile(),'rb+'))
 _pickle.UnpicklingError: pickle data was truncated
```

##### Do you have a suggested fix for the development team?
threading.RLock() in steps

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [ ] 1. If the issue is a defect, is the defect fixed?
- [ ] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [ ] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [ ] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [ ] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by JiaZhou-PU**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #1092 

##### What are the significant changes in functionality due to this change request?
Adding a threading lock to pickled file to avoid pickle data was truncated problem, adding a check if  the path is absolute inside RAVEN parser to avoid the copying same file error.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [ ] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: framework/Steps.py [hunk 1]
```diff
@@ -990,6 +991,9 @@ def _localTakeAstepRun(self,inDictionary):
         ## unpickle the ROM
         fileobj = inDictionary['Input'][i]
         unpickledObj = pickle.load(open(fileobj.getAbsFile(),'rb+'))
```
Comment (alfoa) [inline framework/Steps.py]: you should remove this line if you keep the ones below

File: framework/CodeInterfaces/RAVEN/RAVENparser.py [hunk 2]
```diff
@@ -142,6 +142,8 @@ def findSlaveFiles(self, tree, workingDir):
           moduleToLoad = extModel.attrib['ModuleToLoad']
           if not moduleToLoad.endswith("py"):
             moduleToLoad += ".py"
+          if os.path.isabs(moduleToLoad):
+            continue
```
Comment (alfoa) [inline framework/CodeInterfaces/RAVEN/RAVENparser.py]: what is this for?

File: framework/Steps.py (hunk 1 above)
Comment (JiaZhou-PU) [reply framework/Steps.py]: I thought we should close this PR, it's too slow...

**Comment by alfoa** [comment]
?

**Comment by PaulTalbot-INL** [comment]
This feature appears to be unneeded at this point. Closing for now.

**Pull request closed without merging**